presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp

```diff
@@ -25,6 +24,30 @@ velox::core::PlanNodeId deserializePlanNodeId(const folly::dynamic& obj) {
   return obj["id"].asString();
 }
 
+/// Various state of execution of a PartitionAndSerializeOperator instance.
+enum class ProcessingState {
+  //  Unknown state and can only be valid during
+  // initialization chain or as an default value. If we get to this state while
+  // execution, we likely throw an
+  // error.
+  kUnknown,
+  // Process only the input elements without
+  // consideration to extraneous states like replicateNullAndAny. This state is
+  // visited once for ech input vector.
+  kProcessingInputElements,
+  // Replicate null values until we have filled up the
+  // outputVector_. This state is visited multiple times until we have
+  // replicated all null values.
+  kProcessingNulls,
+  // Replicate 'any' value. This state is visited once in this
+  // operator's lifecycle (at the end), when all the input batches have been
```

Comment:
If this is processed at the end, do we store the original string buffer that holds the value?

This original buffers holds many other rows and consumes memory. Perhaps, we can store just a single string we need.